File: src/PandaPress.Core/Models/Data/Post.cs
using System;
using System.Collections.Generic;

namespace PandaPress.Core.Models.Data
{
    public class Post
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public DateTime PublishDate { get; set; }
        public string Slug { get; set; }
        public bool Published { get; set; }
        public ApplicationUser User { get; set; }
        public string UserId { get; set; }
        public Blog Blog { get; set; }
        public int BlogId { get; set; }
        public List<PostCategory> PostCategories { get; set; } = new List<PostCategory>();
        public List<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: src/PandaPress.Web/ClientApp/app/models/post.ts
import { PostComment } from "./post-comment";
import { PostCategory } from "./post-category";

export class Post {
    constructor() {
        this.comments = new Array();
        this.categories = new Array();
    }
    title: string;
    content: string;
    userDisplayName: string;
    userAbout: string;
    profilePicture: string;
    slug: string;
    comments: PostComment[];
    commentCount: number;
    id: string;
    publishDate: string;
    categories: PostCategory[];
}